From 439658a4af6d6f33bf1b57a8c09d7979eddfdbb8 Mon Sep 17 00:00:00 2001
From: Kriti Pachhandara <kriti.pachhandara@intel.com>
Date: Tue, 2 Dec 2014 14:58:02 +0200
Subject: [PATCH 182/196] atomisp2: Configure yuvpp pipe for soc sensors in
 moorefield

This patch adds support for the copy binary and yuvpp pipe
configuration in the soc sensors use case

Change-Id: I8996185c272a26267a218f2d46780a41286cc8fe
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-11370
Signed-off-by: Kriti Pachhandara <kriti.pachhandara@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/31052
Reviewed-by: Tian Shu Qiu <tian.shu.qiu@intel.com>
Approver: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
---
 .../media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c     | 8 ++++++--
 .../drivers/media/pci/atomisp2/atomisp_driver/atomisp_fops.c     | 5 ++++-
 .../drivers/media/pci/atomisp2/atomisp_driver/atomisp_internal.h | 9 +++++++++
 3 files changed, 19 insertions(+), 3 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
index ad51856..0d47f09 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
@@ -3020,10 +3020,14 @@ int atomisp_get_css_frame_info(struct atomisp_sub_device *asd,
 {
 	struct ia_css_pipe_info info;
 	int pipe_index = atomisp_get_pipe_index(asd, source_pad);
-	int stream_index = (pipe_index == IA_CSS_PIPE_ID_YUVPP) ?
+	int stream_index;
+	if (ATOMISP_SOC_CAMERA(asd))
+		stream_index = atomisp_source_pad_to_stream_id(asd, source_pad);
+	else {
+		stream_index = (pipe_index == IA_CSS_PIPE_ID_YUVPP) ?
 			   ATOMISP_INPUT_STREAM_VIDEO :
 			   atomisp_source_pad_to_stream_id(asd, source_pad);
-
+	}
 	ia_css_pipe_get_info(asd->stream_env[stream_index]
 		.pipes[pipe_index], &info);
 	switch (source_pad) {
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_fops.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_fops.c
index f524791..a2b955a 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_fops.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_fops.c
@@ -488,7 +488,10 @@ int atomisp_qbuffers_to_css(struct atomisp_sub_device *asd)
 		buf_type = atomisp_get_css_buf_type(
 			asd, css_preview_pipe_id,
 			atomisp_subdev_source_pad(&preview_pipe->vdev));
-		if (css_preview_pipe_id == CSS_PIPE_ID_YUVPP)
+		if (ATOMISP_SOC_CAMERA(asd) && css_preview_pipe_id == CSS_PIPE_ID_YUVPP)
+			input_stream_id = ATOMISP_INPUT_STREAM_GENERAL;
+		 /* else for ext isp use case */
+		else if (css_preview_pipe_id == CSS_PIPE_ID_YUVPP)
 			input_stream_id = ATOMISP_INPUT_STREAM_VIDEO;
 		else if (asd->stream_env[ATOMISP_INPUT_STREAM_PREVIEW].stream)
 			input_stream_id = ATOMISP_INPUT_STREAM_PREVIEW;
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_internal.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_internal.h
index 677b6fb..588593b 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_internal.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_internal.h
@@ -169,6 +169,15 @@
 	&& asd->isp->inputs[asd->input_curr].camera_caps-> \
 	   sensor[asd->sensor_curr].stream_num == 1   \
 	&& ATOMISP_CSS_SUPPORT_YUVPP)
+/*
+ * ATOMISP_SOC_CAMERA
+ * This is to differentiate between ext-isp and soc camera in
+ * Moorefield/Baytrail platform.
+ */
+#define ATOMISP_SOC_CAMERA(asd)  \
+	(asd->isp->inputs[asd->input_curr].type == SOC_CAMERA \
+	&& asd->isp->inputs[asd->input_curr].camera_caps-> \
+	   sensor[asd->sensor_curr].stream_num == 1)
 
 #define ATOMISP_DEPTH_SENSOR_STREAMON_COUNT 2
 
-- 
1.9.1

